internal/cli/branding_texts.go

```diff
@@ -90,164 +117,190 @@ func showBrandingTexts(cli *cli, inputs *brandingTextsInputs) func(cmd *cobra.Co
 			return fmt.Errorf("failed to serialize the prompt custom text to JSON: %w", err)
 		}
 
-		cli.renderer.BrandingTextShow(string(brandingTextJSON), prompt, inputs.Language)
+		cli.renderer.BrandingTextShow(string(brandingTextJSON), inputs.Prompt, inputs.Language)
 
 		return nil
 	}
 }
 
-func updateBrandingTextCmd(cli *cli) *cobra.Command {
-	var inputs struct {
-		Language string
-		Body     string
-	}
-
-	cmd := &cobra.Command{
-		Use:   "update",
-		Args:  cobra.ExactArgs(1),
-		Short: "Update the custom texts for a prompt",
-		Long:  "Update the custom texts for a prompt.",
-		Example: `
-auth0 branding texts update <prompt> --language es
-auth0 branding texts update <prompt> -l es`,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			inputs.Body = string(iostream.PipedInput())
-			prompt := args[0]
-			fileName := fmt.Sprintf("%s.%s.json", prompt, inputs.Language)
-			currentBody := make(map[string]interface{})
-			currentBody[textDocsKey] = brandingTextDocsURL(prompt)
-
-			if err := ansi.Waiting(func() error {
-				defaultTranslations := downloadBrandingTextLocale(fileName)
-
-				customTranslations, err := cli.api.Prompt.CustomText(prompt, inputs.Language)
-				if err != nil {
-					return err
-				}
-
-				currentBodyTranslations := mergeBrandingTextLocales(defaultTranslations, customTranslations)
-				for k, v := range currentBodyTranslations {
-					currentBody[k] = v
-				}
-				return nil
-			}); err != nil {
-				return fmt.Errorf("Unable to load custom text for prompt %s and language %s: %w", prompt, inputs.Language, err)
-			}
+func updateBrandingText(cli *cli, inputs *brandingTextsInputs) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		inputs.Prompt = args[0]
+		inputs.Body = string(iostream.PipedInput())
 
-			currentBodyStr, err := marshalBrandingTextBody(currentBody)
-			if err != nil {
-				return err
-			}
+		brandingTextToEdit, err := fetchBrandingTextContentToEdit(cli, inputs)
+		if err != nil {
+			return fmt.Errorf("failed to fetch branding text content to edit: %w", err)
+		}
 
-			if err := ruleScript.OpenEditor(
-				cmd,
-				&inputs.Body,
-				currentBodyStr,
-				fileName,
-				cli.promptTextEditorHint,
-			); err != nil {
-				return fmt.Errorf("Failed to capture input from the editor: %w", err)
-			}
+		editedBrandingText, err := fetchEditedBrandingTextContent(cmd, cli, inputs, brandingTextToEdit)
```

Comment:
🔧 🤔 An e2e flow is really really hard to test cuz of this func that opens the editor. In order to mock that part we need to better refactor the code to make it easier to test. I decided to defer that to another time and just focus on fixing this cmd first. Happy to discuss ideas tho.

Reply:
We originally added E2E tests because at such an early stage, a bunch of unit tests would be more of a hindrance when moving fast and changing lots of things. I would expect that, for v1.0.0, to have lots of unit tests and consider which E2E tests we need, if any at all.